consistent-sampling/src/main/java/io/opentelemetry/contrib/sampler/consistent56/ConsistentFixedThresholdSampler.java

```diff
@@ -41,7 +49,18 @@ public String getDescription() {
   }
 
   @Override
-  protected long getThreshold(long parentThreshold, boolean isRoot) {
-    return threshold;
+  public SamplingIntent getSamplingIntent(
+      Context parentContext,
+      String name,
+      SpanKind spanKind,
+      Attributes attributes,
+      List<LinkData> parentLinks) {
+
+    return () -> {
+      if (threshold == getMaxThreshold()) {
+        return getInvalidThreshold();
+      }
```

Comment:
Please comment, I had to go review https://github.com/open-telemetry/opentelemetry-java-contrib/blob/main/consistent-sampling/src/main/java/io/opentelemetry/contrib/sampler/consistent56/ConsistentSamplingUtil.java to understand that `getMaxThreshold()` is what I was expecting `getInvalidThreshold()` to be. 

Why does the Util package distinguish Max and Invalid? (Why aren't they a single value, maybe?)

Reply:
This probably requires a cleanup. The prototype currently uses `long` rather than hex `String` to represent the thresholds, but the proposed specs allow both implementations. I'd love to see the same logic and behavior for all the new samplers regardless of the choice of the threshold representation. However, the issue is with the MaxThreshold, which is really an out-of-range value. Representing the maximum threshold as a hex string is not useful, as it would be a 15-character string and it would break lexicographical comparisons.
Therefore I opted into requiring that `getSamplingIntent` always provides a _valid_ threshold (which is smaller than 2^56), or an _invalid_ one, which is a special value easy to test for.